cmd/copyUtil.go

```diff
@@ -55,26 +54,21 @@ func (copyHandlerUtil) numOfWildcardInURL(url url.URL) int {
 	return strings.Count(url.String(), wildCard)
 }
 
-// isIPEndpointStyle checkes if URL's host is IP, in this case the storage account endpoint will be composed as:
-// http(s)://IP(:port)/storageaccount/share(||container||etc)/...
-// TODO: Remove this, it can be replaced by SDK's native support for IP endpoint style.
-func (util copyHandlerUtil) isIPEndpointStyle(url url.URL) bool {
-	return net.ParseIP(url.Host) != nil
-}
-
 // checks if a given url points to a container, as opposed to a blob or prefix match
 func (util copyHandlerUtil) urlIsContainerOrShare(url *url.URL) bool {
-	// When it's IP endpoint style, if the path contains more than two "/", then it means it points to a blob, and not a container.
-	// When it's not IP endpoint style, if the path contains more than one "/", then it means it points to a blob, and not a container.
-	numOfSlashes := strings.Count(url.Path[1:], "/")
-	isIPEndpointStyle := util.isIPEndpointStyle(*url)
-
-	if (!isIPEndpointStyle && numOfSlashes == 0) || (isIPEndpointStyle && numOfSlashes == 1) {
-		return true
-	} else if ((!isIPEndpointStyle && numOfSlashes == 1) || (isIPEndpointStyle && numOfSlashes == 2)) && strings.HasSuffix(url.Path, "/") { // this checks if container_name/ was given
-		return true
+	if azblob.NewBlobURLParts(*url).IPEndpointStyleInfo.AccountName == "" {
+		//Typical endpoint style
+		//If there's no slashes after the first, it's a container.
+		//If there's a slash on the end, it's a virtual directory/container.
+		//Otherwise, it's just a blob.
+		return strings.HasSuffix(url.Path, "/") || strings.Count(url.Path[1:], "/") == 0
+	} else {
+		//URL endpoint style: https://IP:port/accountname/container
```

Comment:
This is an example of IP endpoint style, but its labelled URL style.  I think its just the comment that's wrong. Should say `// IP endpoint style: https://IP:port/accountname/container`. Do you agree?

Reply:
Yeah, my mistake when writing that